ister.py

```diff
@@ -640,11 +654,49 @@ def post_install_nonchroot(template, target_dir):
 
     if not template.get("PostNonChroot"):
         return
-    LOG.info("Running post scripts")
+    LOG.info("Running post non-chroot scripts")
     for script in template["PostNonChroot"]:
         run_command(script + " {}".format(target_dir))
 
 
+def post_install_nonchroot_shell(template, target_dir):
+    """Run non chroot post install commands
+
+    The mount root for the install is passed as the ISTER_CHROOT environment
+    variable.
+    """
+    if not template.get("PostNonChrootShell"):
+        return
+    LOG.info("Running post non-chroot commands")
+    script_env = os.environ
+    script_env["ISTER_CHROOT"] = target_dir
+    for cmdl in template["PostNonChrootShell"]:
+        run_command(cmdl, shell=True, environ=script_env)
+
+
+def post_install_chroot(template, target_dir):
+    """Run chroot post install scripts
+
+    All post scripts must be executable.
+    """
+    if not template.get("PostChroot"):
+        return
+    LOG.info("Running post scripts")
+    with ChrootOpen(target_dir) as _:
+        for script in template["PostChroot"]:
+            run_command(script + " {}".format(target_dir))
```

Comment:
Is it documented that the PostChroot script must accept a single target_dir argument?

Reply:
I guess I would have the same question for the PostNonChroot script - I've never made use of these myself.

Reply:
Ah PostChroot shouldn't get the target_dir passed in that was an oversight. And sadly there is not much documentation at all on the script requirements. Making the README useful is quite a work item that needs to be looked at.